velox/functions/prestosql/window/FirstLastValue.cpp

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "velox/common/base/Exceptions.h"
+#include "velox/exec/WindowFunction.h"
+#include "velox/expression/FunctionSignature.h"
+#include "velox/vector/FlatVector.h"
+
+namespace facebook::velox::window::prestosql {
+
+namespace {
+
+enum class ValueType {
+  kFirst,
+  kLast,
+};
+
+template <ValueType TValue, typename T>
```

Comment:
Template parameter T is not used. Let's remove it.

Let's also remove createFirstLastValueFunction helper function and just use 

```
return std::make_unique<FirstLastValueFunction<TValue>>(
            args, resultType, pool);
```

Same for NthValueFunction.

Reply:
Updated, thanks!